comma-dangle reports incorrect position in array with ternary item

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:** `3.7.0`
* **Node Version:** `4.2.2`
* **npm Version:** `3.10.7`

**What parser (default, Babel-ESLint, etc.) are you using?**

`babel-eslint@7.0.0`

**Please show your full configuration:**

<!-- Paste your configuration below: -->
```
module.exports = {
  "extends": "eslint-config-airbnb",
  "env": {
    "browser": true,
    "node": true,
    "mocha": true
  },
  "parser": "babel-eslint",
  "rules": {
    "no-unused-vars": "warn",
    "no-alert": "warn",
    "no-console": "warn",
    "no-use-before-define": "warn",
    "no-var": "warn",
    "block-scoped-var": "warn",
    "guard-for-in": "off",
    "no-shadow": "off",
    "no-param-reassign": "off",

    "indent": [ "warn", 2, { "SwitchCase": 1 } ],
    "padded-blocks": "off",
    "spaced-comment": "warn",
    "comma-dangle": [ "warn", "always-multiline" ],
    "space-in-parens": "off",
    "curly": [ "off", "all" ],
    "array-bracket-spacing": [ "warn", "always" ],
    "object-curly-spacing": [ "warn", "always" ],
    "computed-property-spacing": "off",
    "brace-style": [ "warn", "stroustrup", { "allowSingleLine": true } ],
    "no-trailing-spaces": [ "warn", { "skipBlankLines": true } ],
    "linebreak-style": [ "warn", "unix" ],
    "no-multiple-empty-lines": [ "warn", { "max": 2, "maxEOF": 1 } ],
    "eol-last": [ "error", "unix" ],
    "id-length": [ "warn", { "min": 2, "exceptions": [ "_", "$", "i", "j", "k", "x", "y", "e" ] } ],
    "camelcase": "warn",
    "func-names": "warn",
    "keyword-spacing": "warn",
    "space-before-blocks": "warn",
    "space-before-function-paren": [ "warn", { "anonymous": "always", "named": "never" } ],
    "quotes": [ "warn", "single", "avoid-escape" ],
    "no-multi-spaces": [ "warn", { "exceptions": { "VariableDeclarator": true, "ImportDeclaration": true } } ],
    "dot-notation": "off",
    "prefer-template": "off",
    "prefer-arrow-callback": "warn",
    "max-len": [ "warn", 250, 4, { "ignoreComments": true } ],
    "arrow-parens": [ "warn", "always" ],
    "arrow-body-style": "off",
    "object-shorthand": "off",
    "no-case-declarations": "warn",
    "no-nested-ternary": "off",
    "global-require": "off",
    "no-underscore-dangle": "off",
    "no-useless-concat": "off",
    "no-mixed-operators": "off",
    "no-bitwise": "off",

    "react/prop-types": "warn",
    "react/forbid-prop-types": "off",
    "react/no-unused-prop-types": "off",
    "react/react-in-jsx-scope": "error",
    "react/jsx-filename-extension": [ "warn", { "extensions": [ ".js", ".jsx" ] } ],
    "react/jsx-uses-react": "error",
    "react/jsx-uses-vars": "error",
    "react/jsx-quotes": "off",
    "react/jsx-closing-bracket-location": "warn",
    "react/jsx-space-before-closing": "warn",
    "react/jsx-curly-spacing": "off",
    "react/jsx-indent": "off",
    "react/self-closing-comp": "warn",
    "react/no-multi-comp": "off",
    "react/sort-comp": "off",
    "react/prefer-stateless-function": "warn",
    "react/no-children-prop": "off",
    "jsx-quotes": "error",
    "jsx-a11y/img-uses-alt": "off",

    "import/default": "off",
    "import/no-duplicates": "off",
    "import/named": "off",
    "import/namespace": "off",
    "import/newline-after-import": "off",
    "import/imports-first": "off",
    "import/no-unresolved": "error",
    "import/no-named-as-default": "error",
    "import/extensions": [ "warn", "always", { "": "never", "js": "never" } ],
  },
  "plugins": [
    "react",
    "import",
  ],
  "settings": {
    "import/ignore": [
      "node_modules",
      "\\.(scss|less|css)$",
    ],
    "import/resolver": {
      "node": {
        "moduleDirectory": [
          "./src",
          "./api",
          "./common",
        ],
      },
      "webpack": {
        "config": "./webpack/configForEslintImportResolver.js",
      },
    },
  },
  "globals": {
    "__SERVER_ENV__": true,
    "__DEVELOPMENT__": true,
    "__SERVER__": true,
    "__SERVER_DEVTOOLS__": true,
    "__SERVER_DISABLE_SERVERSIDE_RENDERING__": true,
    "__CLIENT__": true,
    "__CLIENT_DEVTOOLS__": true,
    "__LOCALHOST__": true,
    "__FAST_DEVELOPMENT__": true,
  },
};
```

**What did you do? Please include the actual source code causing the issue.**

Reduced test case:
<!-- Paste the source code below: -->
```js
let xxx; let yyy;
const zzz = [
  (xxx
    ? yyy
    : ''   // <<< reports that the comma is missing here
  )   // <<< should report the missing comma here
];
zzz.toString();  // to prevent unused variable warning in the reduced test case

```

**What did you expect to happen?**

Correct positioning of the warning (after the closing paren where the comma-dangle is expected).

**What actually happened? Please include the actual, raw output from ESLint.**

```
  5:13  warning  Missing trailing comma  comma-dangle
```

--- a/lib/rules/comma-dangle.js
+++ b/lib/rules/comma-dangle.js
@@ -155,7 +155,7 @@ module.exports = {
             if (trailingToken.value !== ",") {
                 context.report({
                     node: lastItem,
-                    loc: lastItem.loc.end,
+                    loc: penultimateToken.loc.end,
                     message: MISSING_MESSAGE,
                     fix(fixer) {
                         return fixer.insertTextAfter(penultimateToken, ",");

--- a/tests/lib/rules/comma-dangle.js
+++ b/tests/lib/rules/comma-dangle.js
@@ -527,6 +527,33 @@ ruleTester.run("comma-dangle", rule, {
             ]
         },
         {
+
+            // https://github.com/eslint/eslint/issues/7291
+            code:
+            "var foo = [\n" +
+            "  (bar\n" +
+            "    ? baz\n" +
+            "    : qux\n" +
+            "  )\n" +
+            "];",
+            output:
+            "var foo = [\n" +
+            "  (bar\n" +
+            "    ? baz\n" +
+            "    : qux\n" +
+            "  ),\n" +
+            "];",
+            options: [ "always" ],
+            errors: [
+                {
+                    message: "Missing trailing comma.",
+                    type: "ConditionalExpression",
+                    line: 5,
+                    column: 4
+                }
+            ]
+        },
+        {
             code: "var foo = { bar: 'baz', }",
             output: "var foo = { bar: 'baz' }",
             options: [ "always-multiline" ],

--- a/tests/lib/rules/comma-dangle.js
+++ b/tests/lib/rules/comma-dangle.js
@@ -477,6 +477,56 @@ ruleTester.run("comma-dangle", rule, {
             ]
         },
         {
+            code:
+            "var foo = [\n" +
+            "  bar,\n" +
+            "  (\n" +
+            "    baz\n" +
+            "  )\n" +
+            "];",
+            output:
+            "var foo = [\n" +
+            "  bar,\n" +
+            "  (\n" +
+            "    baz\n" +
+            "  ),\n" +
+            "];",
+            options: [ "always" ],
+            errors: [
+                {
+                    message: "Missing trailing comma.",
+                    type: "Identifier",
+                    line: 5,
+                    column: 4
+                }
+            ]
+        },
+        {
+            code:
+            "var foo = {\n" +
+            "  foo: 'bar',\n" +
+            "  baz: (\n" +
+            "    qux\n" +
+            "  )\n" +
+            "};",
+            output:
+            "var foo = {\n" +
+            "  foo: 'bar',\n" +
+            "  baz: (\n" +
+            "    qux\n" +
+            "  ),\n" +
+            "};",
+            options: [ "always" ],
+            errors: [
+                {
+                    message: "Missing trailing comma.",
+                    type: "Property",
+                    line: 5,
+                    column: 4
+                }
+            ]
+        },
+        {
             code: "var foo = { bar: 'baz', }",
             output: "var foo = { bar: 'baz' }",
             options: [ "always-multiline" ],

